cdap-ui/app/cdap/components/EntityListView/index.js

```diff
@@ -125,6 +127,23 @@ class EntityListView extends Component {
     this.getQueryObject = this.getQueryObject.bind(this);
     this.handlePageChange = this.handlePageChange.bind(this);
     this.setAnimationDirection = this.setAnimationDirection.bind(this);
+    this.eventEmitter = ee(ee);
+    this.refreshSearchByCreationTime = this.refreshSearchByCreationTime.bind(this);
+    this.eventEmitter.on(globalEvents.APPUPLOAD, this.refreshSearchByCreationTime);
+    this.eventEmitter.on(globalEvents.STREAMCREATE, this.refreshSearchByCreationTime);
+    this.eventEmitter.on(globalEvents.PUBLISHPIPELINE, this.refreshSearchByCreationTime);
+  }
+
+  refreshSearchByCreationTime() {
+    this.setState({
+      sortObj: this.sortOptions[4]
+    }, this.search.bind(this));
+  }
```

Comment:
should we add a check in here to only refresh the entity list if the user is on page 1 only? or is the expectation is that it will always refresh to page 1?